packages/cli/src/commands/deploy.ts

```diff
@@ -83,4 +87,72 @@ export default class Deploy extends AuthCommand {
       }
     }
   }
+
+  private formatPreview (previewData: any, project: Project): string {
+    // Current format of the data is: { checks: { logical-id-1: 'UPDATE' }, groups: { another-logical-id: 'CREATE' } }
+    // We convert it into update: [{ logicalId, resourceType, construct }, ...], create: [], delete: []
+    // This makes it easier to display.
+    const updating = []
+    const creating = []
+    const deleting = []
+    for (const [resourceType, resourceStatuses] of Object.entries(previewData?.diff ?? {})) {
+      if (resourceType === AlertChannelSubscription.__checklyType) {
+        // Don't report changes to alert channel subscriptions.
+        // User's don't create these directly, so it's more intuitive to consider it as part of the check.
+        continue
+      }
+      for (const [logicalId, resourceStatus] of Object.entries(resourceStatuses ?? {})) {
+        if (resourceStatus === 'UPDATE') {
+          const construct = project.data[resourceType as keyof ProjectData][logicalId]
+          updating.push({ resourceType, logicalId, construct })
+        } else if (resourceStatus === 'CREATE') {
+          const construct = project.data[resourceType as keyof ProjectData][logicalId]
+          creating.push({ resourceType, logicalId, construct })
+        } else if (resourceStatus === 'DELETE') {
+          // Since the resource is being deleted, the construct isn't in the project.
+          deleting.push({ resourceType, logicalId })
+        }
+      }
+    }
+    // Sort the entries by sourceFile, then resource type, then name (if it has a name).
+    // Having some order will make the output easier to read.
+    const compareEntries = (a: any, b: any) =>
+      a.resourceType.localeCompare(b.resourceType) ||
+      a.construct.getSourceFile().localeCompare(b.construct.getSourceFile()) ||
+      (a.construct.name ?? a.logicalId).localeCompare(b.construct.name ?? b.logicalId)
+    updating.sort(compareEntries)
+    creating.sort(compareEntries)
+    deleting.sort((a, b) => a.resourceType.localeCompare(b.resourceType) || a.logicalId.localeCompare(b.logicalId))
+
+    const output = []
+    if (creating.length) {
+      output.push(chalk.bold.green('Create:'))
+      for (const { logicalId, construct } of creating) {
+        const name = (construct as any).name
+        output.push(`    ${construct.constructor.name}: ${construct.getSourceFile()} > ${name ?? logicalId}`)
+      }
+      output.push('')
+    }
+    if (deleting.length) {
+      output.push(chalk.bold.red('Delete:'))
+      const prettyResourceTypes: any = {
```

Comment:
Why any? Record<string, string>?

Reply:
Without the `any`, then there's a TS compile error on line 144 - `resourceType` doesn't have type `keyof prettyResourceTypes`.

I can update it to `Record<string, string>`, that should also solve it.

Reply:
Please avoid using `any` as much as you can